ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteSessionImpl.java

```diff
@@ -0,0 +1,88 @@
+package org.hibernate.ogm.datastore.ignite.impl;
+
+import java.util.List;
+
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.event.spi.EventSource;
+import org.hibernate.internal.CriteriaImpl;
+import org.hibernate.ogm.OgmSessionFactory;
+import org.hibernate.ogm.datastore.ignite.loader.criteria.OgmCriteriaLoader;
+import org.hibernate.ogm.datastore.ignite.loader.criteria.impl.CriteriaCustomQuery;
+import org.hibernate.ogm.exception.NotSupportedException;
+import org.hibernate.ogm.hibernatecore.impl.OgmSessionImpl;
+import org.hibernate.resource.transaction.TransactionCoordinator;
+import org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorImpl;
+
+public class IgniteSessionImpl extends OgmSessionImpl {
```

Comment:
I think it would be best to separate the Criteria part from the implementation of the dialect.
It would be a new feature for Hibernate OGM and it requires some additional discussion.